### ISSUE TITLE
Tổng hợp các lỗi sai chính tả

### ISSUE THREAD

**Issue by ZuzooVn** [tiepvupsu/ebookML_src#17]
Hi a Tiệp, @tiepvupsu 

Em tổng hợp lại 1 số lỗi chính tả như sau, các lỗi sẽ được tiếp tục update.

--------------------------

- Trang 2 
<img width="485" alt="screen shot 2018-01-19 at 4 00 24 pm" src="https://user-images.githubusercontent.com/1429230/35142803-0ccb91dc-fd32-11e7-9f2d-502a52653fb2.png">

cùng => cũng

--------------------------

- Trang 3

<img width="435" alt="screen shot 2018-01-19 at 4 05 14 pm" src="https://user-images.githubusercontent.com/1429230/35142970-9186d7a6-fd32-11e7-8060-6cecc95ffb02.png">

được cho => được vẽ

--------------------------

- Trang 4

<img width="239" alt="screen shot 2018-01-19 at 4 05 54 pm" src="https://user-images.githubusercontent.com/1429230/35143002-af5bbc10-fd32-11e7-9bc8-33468fa63ea9.png">
 
deeep learning => deep learning

<img width="246" alt="screen shot 2018-01-19 at 4 06 48 pm" src="https://user-images.githubusercontent.com/1429230/35143035-caf5e6bc-fd32-11e7-9403-48bf7dc4090f.png">

scikit-learng => scikit-learn

<img width="226" alt="screen shot 2018-01-19 at 4 07 32 pm" src="https://user-images.githubusercontent.com/1429230/35143068-e4725ddc-fd32-11e7-8511-8bbca65100d5.png">

nhưng thư viện => những thư viện

--------------------------

- Trang 7

<img width="212" alt="screen shot 2018-01-19 at 4 08 30 pm" src="https://user-images.githubusercontent.com/1429230/35143101-0c6b13f6-fd33-11e7-97aa-af84889c328e.png">

bản quyển => bản quyền

<img width="204" alt="screen shot 2018-01-19 at 4 09 14 pm" src="https://user-images.githubusercontent.com/1429230/35143125-21460b8c-fd33-11e7-8ec7-a26f3d7fb742.png">

nếu rõ tên => nêu rõ tên

--------------------------

- Trang 12

<img width="365" alt="screen shot 2018-01-19 at 4 10 05 pm" src="https://user-images.githubusercontent.com/1429230/35143152-3ccaad4a-fd33-11e7-9dea-427f827069bf.png">

dầu phẩy => dấu phẩy

--------------------------

- Trang 15

<img width="306" alt="screen shot 2018-01-19 at 4 53 55 pm" src="https://user-images.githubusercontent.com/1429230/35145163-6f6d1e1c-fd39-11e7-8d14-9898a8b6c5a8.png">

=> không (bỏ dấu gạch trên)

--------------------------

- Trang 36

<img width="339" alt="screen shot 2018-01-19 at 5 04 30 pm" src="https://user-images.githubusercontent.com/1429230/35145574-e382a686-fd3a-11e7-9e84-f2f106701ae9.png">

Bẵng => Bằng

--------------------------

- Trang 52

<img width="345" alt="screen shot 2018-01-19 at 4 10 54 pm" src="https://user-images.githubusercontent.com/1429230/35143185-5e462044-fd33-11e7-8a3c-a6b7d084c3b7.png">

tập huấn luyên => tập huấn luyện

--------------------------

- Trang 53

<img width="310" alt="screen shot 2018-01-19 at 4 11 52 pm" src="https://user-images.githubusercontent.com/1429230/35143215-7b8dcc10-fd33-11e7-9c98-5638855ab9c9.png">

có điều kiên => có điều kiện

<img width="437" alt="screen shot 2018-01-19 at 4 12 15 pm" src="https://user-images.githubusercontent.com/1429230/35143231-8b42784a-fd33-11e7-9721-b4d74a9f6e2e.png">

xảy ra đồng thời xảy ra => đồng thời xảy ra

--------------------------

- Trang 64

<img width="338" alt="screen shot 2018-01-19 at 4 13 21 pm" src="https://user-images.githubusercontent.com/1429230/35143276-b033b100-fd33-11e7-800e-045ab13403b5.png">

machine learing => machine learning

--------------------------

- Trang 68

<img width="286" alt="screen shot 2018-01-19 at 4 14 31 pm" src="https://user-images.githubusercontent.com/1429230/35143339-de77db9a-fd33-11e7-8b5b-c620f53fc64a.png">

dữ liệu vào X => dữ liệu đầu vào X

<img width="859" alt="screen shot 2018-01-19 at 4 15 23 pm" src="https://user-images.githubusercontent.com/1429230/35143402-fd931b66-fd33-11e7-8777-000e59ecdd37.png">

Vì dụ => Ví dụ
xunh quanh => xung quanh

--------------------------

- Trang 69

<img width="858" alt="screen shot 2018-01-19 at 4 17 20 pm" src="https://user-images.githubusercontent.com/1429230/35143477-47bf88dc-fd34-11e7-8814-0f0510961ffd.png">

đề tìm tòi => để tìm tòi
thằng tất cả => thắng tất cả

--------------------------

- Trang 70

<img width="298" alt="screen shot 2018-01-19 at 4 18 30 pm" src="https://user-images.githubusercontent.com/1429230/35143516-68c29d8a-fd34-11e7-9e87-9766084a19a8.png">

mô mô hình => mô hình

- Trang 71

<img width="293" alt="screen shot 2018-01-19 at 4 19 04 pm" src="https://user-images.githubusercontent.com/1429230/35143568-858a18d0-fd34-11e7-8257-74c4a2b35e0e.png">

Điều này là cần thiết các phép toán => Điều này là cần thiết do các phép toán

--------------------------

- Trang 72

<img width="279" alt="screen shot 2018-01-19 at 4 20 06 pm" src="https://user-images.githubusercontent.com/1429230/35143615-a93e695c-fd34-11e7-8820-169d8b5c2dcb.png">

dử liệu => dữ liệu

```
Có hai khối có nền màu lục cần được thiết kế:
Khối thứ nhất, Feature Extraction,
```
=> Phần này đang liệt kê rõ ra 2 khối cần được thiết kế, tuy nhiên khối thứ 2 không được nhắc đến rõ ràng, chỉ được nêu qua qua ở mục cuối của phần `6.2.1`, do đó nếu độc giả đọc lướt qua, thì ko biết được khối thứ 2 đóng vai trò gì. Em nghĩ nên có 1 phần giải thích về khối thứ 2 này.

--------------------------

- Trang 74 

<img width="293" alt="screen shot 2018-01-19 at 4 20 53 pm" src="https://user-images.githubusercontent.com/1429230/35143638-c16791f2-fd34-11e7-8b03-62160fda46a9.png">

extracted features => extracted feature (để phù hợp với extracted feature ở dòng trên)

<img width="236" alt="screen shot 2018-01-19 at 4 22 23 pm" src="https://user-images.githubusercontent.com/1429230/35144056-fb065d34-fd35-11e7-8cf6-5ce196c1ad29.png">

nhờ và việc các hai khối phía trên  => nhờ vào việc hai khối phía trên

--------------------------

- Trang 75

<img width="461" alt="screen shot 2018-01-19 at 4 31 54 pm" src="https://user-images.githubusercontent.com/1429230/35144157-4b34285e-fd36-11e7-8e6d-a173f27f5a73.png">

vector x1 ban đầu => vector x0 ban đầu

<img width="310" alt="screen shot 2018-01-19 at 4 32 46 pm" src="https://user-images.githubusercontent.com/1429230/35144194-69b4ed5e-fd36-11e7-8408-e6f0a93cbddd.png">

trình bay => trình bày

--------------------------

- Trang 76

<img width="827" alt="screen shot 2018-01-19 at 4 33 27 pm" src="https://user-images.githubusercontent.com/1429230/35144226-821ed3dc-fd36-11e7-9f21-adf3faea74c3.png">

phẩn tử => phần tử
từ điền => từ điển

--------------------------

- Trang 77

<img width="152" alt="screen shot 2018-01-19 at 4 34 22 pm" src="https://user-images.githubusercontent.com/1429230/35144266-a3604ea4-fd36-11e7-8eb1-0b31b9d0bc77.png">

lục => xanh lục (cho phù hợp với danh sách màu ở trên)

<img width="227" alt="screen shot 2018-01-19 at 4 35 24 pm" src="https://user-images.githubusercontent.com/1429230/35144314-c7db3172-fd36-11e7-9e46-7e59532bdd39.png">

say mạc => sa mạc

--------------------------

- Trang 78

<img width="293" alt="screen shot 2018-01-19 at 4 35 58 pm" src="https://user-images.githubusercontent.com/1429230/35144341-dc2079d0-fd36-11e7-9bb3-df3fc99fca35.png">

hai patches => hai patch

--------------------------

- Trang 80

<img width="215" alt="screen shot 2018-01-19 at 4 36 49 pm" src="https://user-images.githubusercontent.com/1429230/35144392-f88bcff2-fd36-11e7-9e21-78af81cbb95b.png">

classe => class

<img width="281" alt="screen shot 2018-01-19 at 4 37 24 pm" src="https://user-images.githubusercontent.com/1429230/35144420-0c2cc322-fd37-11e7-9ecf-4ec2ffaac890.png">

bài toàn => bài toán

--------------------------

- Trang 81

<img width="194" alt="screen shot 2018-01-19 at 4 39 38 pm" src="https://user-images.githubusercontent.com/1429230/35144529-62bc476c-fd37-11e7-87ec-c0b0c115f109.png">

nên nên => nên

--------------------------

- Trang 85

<img width="146" alt="screen shot 2018-01-19 at 4 40 23 pm" src="https://user-images.githubusercontent.com/1429230/35144564-77651054-fd37-11e7-8fd8-4db9d6716a29.png">

tính trunh bình => tính trung bình

--------------------------

- Trang 86

<img width="427" alt="screen shot 2018-01-19 at 4 40 59 pm" src="https://user-images.githubusercontent.com/1429230/35144609-977362c4-fd37-11e7-95f6-a7228b297db4.png">

một => 1 (nên dùng số cho thống nhất do ở bên trên có viết x0 =1)

--------------------------

- Trang 91

<img width="176" alt="screen shot 2018-01-19 at 4 42 30 pm" src="https://user-images.githubusercontent.com/1429230/35144669-c5136422-fd37-11e7-80b6-920c02426f4b.png">

một môt hình => một mô hình 

--------------------------

- Trang 94

<img width="256" alt="screen shot 2018-01-19 at 4 43 15 pm" src="https://user-images.githubusercontent.com/1429230/35144694-dcd0d1bc-fd37-11e7-9cef-99fdb0e3f9cb.png">

đai lượng => đại lượng

--------------------------

- Trang 95

<img width="225" alt="screen shot 2018-01-19 at 4 43 48 pm" src="https://user-images.githubusercontent.com/1429230/35144722-f66304d8-fd37-11e7-970c-6c17ed35485e.png">

thường đường => thường được

--------------------------

- Trang 96

<img width="171" alt="screen shot 2018-01-19 at 4 44 28 pm" src="https://user-images.githubusercontent.com/1429230/35144743-0bad73aa-fd38-11e7-982e-8c0dbc31b5f0.png">

validation set set => validation set

**Comment by ZuzooVn** [comment]
Em ko có mail báo cập nhật bản January 10, 2018, nên có 1 vài lỗi ở trên đã được chỉnh sửa rồi.
Em sẽ chuyển sang report cho bản January 10, 2018 này :)

**Comment by tiepvupsu** [comment]
Cảm ơn em. Đúng là lỗi chính tả nhiều vô số. Chắc anh phải làm cái plugin trong Sublime cho kiểm tra lỗi chính tả :D.

**Comment by ZuzooVn** [comment]
https://github.com/titoBouzout/Dictionaries
có hỗ trợ tiếng Việt đây anh

**Comment by tiepvupsu** [comment]
Cảm ơn em. 
Anh vừa thử, cái này chỉ ở mức độ 1 tiếng thôi. Ví dụ 'say mạc' nó vẫn không thể nhận được là sai. Chắc phải có tí NLP vào nữa nó mới có thể nhận được cả cụm. 

Good to know, anyway.

**Comment by ZuzooVn** [comment]
Bản January 10, 2018

- Trang 106:

print("Accuracy of `10NN` with major voting: %.2f %%" %(100 * accuracy_score(y_test, y_pred)))
=>
print("Accuracy of `7NN` with major voting: %.2f %%" %(100 * accuracy_score(y_test, y_pred)))


- Trang 107:

Accuracy of `10NN` with major voting: 93.85 %
=> 
Accuracy of `7NN` with major voting: 93.85 %

- Trang 109:

Trong Hình 9.3, chữ đè lên ảnh.

- Trang 110:

từ cluster sẽ được sử dụng thay thế cho nhóm/`cum`
=>  
từ cluster sẽ được sử dụng thay thế cho nhóm/`cụm`

- Trang 114:

3. Nếu việc phân `nhớm` dữ liệu => 3. Nếu việc phân `nhóm` dữ liệu

- Trang 118:

từ 0 `đên` 9 => từ 0 `đến` 9

Các pixel màu đen có giá trị bằng `không ,` => Các pixel màu đen có giá trị bằng `không,`

Để `download` MNIST => Để `tải về` MNIST 

- Trang 121:

sau dó ta sẽ `chon chọn` cluster => sau dó ta sẽ `chọn` cluster

mỗi điểm ảnh sẽ được biểu diễn `bới` => mỗi điểm ảnh sẽ được biểu diễn `bởi` 

- Trang 122:

chúng ta có ba `clusters` => chúng ta có ba `cluster`


- Trang 123:

trong Hình 10.9:  `k = 5` => `K = 5`

- Trang 123:

Cá́c cluster cần có `só` => Cá́c cluster cần có `số` 

- Trang 127:

`dầu` bằng thứ ba xảy ra => `dấu` bằng thứ ba xảy ra


- Trang 128:

`Gaussian naive Bayes, multinomial naive Bayes, và Bernoulli Naive` 
=>
`Gaussian Naive Bayes, Multinomial Naive Bayes và Bernoulli Naive`

`11.2.1 Gaussian naive Bayes`
=>
`11.2.1 Gaussian Naive Bayes`

- Trang 129:

`11.2.2 Multinomial naive Bayes`
=>
`11.2.2 Multinomial Naive Bayes`

- Trang 130:

`multinomial Naive Bayes hoặc Bernoulli naive Bayes`
=>
`Multinomial Naive Bayes hoặc Bernoulli Naive Bayes`

- Trang 141:

Xét các hàm số một `biết`
=>
Xét các hàm số một `biến`


là `nên` tảng cho rất nhiều phương pháp tối ưu
=>
là `nền` tảng cho rất nhiều phương pháp tối ưu

- Trang 142:

(x2, it2) = myGD1(5, `1`)
=>
(x2, it2) = myGD1(5, `.1`)

- Trang 147:

việc chọn learning rate `họp` lý là rất quan trọng
=>
việc chọn learning rate `hợp` lý là rất quan trọng

- Trang 149:

Hình 12.8: `GD with momentum` => `GD với momentum`

Làm thế nào để biểu diễn momentum `đưới` dạng toán học?
=>
Làm thế nào để biểu diễn momentum `dưới` dạng toán học?

`Sử` khác nhau giữa GD thông thường
=>
`Sự` khác nhau giữa GD thông thường

- Trang 150:

Hình 12.10: `GDt` với momentum. => `GD` với momentum.

- Trang 153:

Hình 12.12  `(b) Nesterov accelerated gradient.` => `(b) GD với NAG` (cho phù hợp với "Chúng ta cùng áp dụng cả GD với momentum và GD với NAG cho bài toán linear regression đề cập ở trên.")

**Comment by tiepvupsu** [comment]
Cảm ơn em rất nhiều. 

Anh đã sửa gần hết, chỉ còn các lỗi về hình (anh tìm lại chưa thấy source code đâu). 

Còn các lỗi:
Trang 109:
Trong Hình 9.3, chữ đè lên ảnh.

Trang 123:
trong Hình 10.9: k = 5 => K = 5

Trang 149:
Hình 12.8: GD with momentum => GD với momentum

Các issues về `naive` -> `Naive` anh xin không sửa. Trên blog anh đều dùng viết hoa cho các tên thuật toán, nhưng tham khảo nhiều sách (Bishop 2006) thì tên các thuật toán khi viết đầy đủ đều viết thường trừ tên riêng.

**Comment by ZuzooVn** [comment]
- Trang 156

`Thuât` toán này được thiết kế => `Thuật` toán này được thiết kế

Hình 13.1a tương ứng với `các điểm màu xanh` => Hình 13.1a tương ứng với `tập các điểm màu xanh`

ta đang giả sử rằng tồn tại một `siểu phẳng` như vậy => ta đang giả sử rằng tồn tại một `siêu phẳng` như vậy

- Trang 158

đây là một `hàm số này` rời rạc => đây là một `hàm số` rời rạc

- Trang 166

các điểm càng xa ngưỡng về phía `phải` có giá trị càng gần 1 => các điểm càng xa ngưỡng về phía `bên phải` có giá trị càng gần 1

- Trang 168

trong dấu `argmin` của (14.7) => trong dấu `argmax` của (14.7)

- Trang 174

ta khai báo các thư viện và `download` => ta khai báo các thư viện và `tải về`

- Trang 176

các bộ lớp nhị `phần` => các bộ lớp nhị `phân`

- Trang 177

hoặc `one-against-all) .` => hoặc `one-against-all).`

- Trang 178

tổng quát lên `nhớ` các phép tính với ma trận => tổng quát lên các phép tính với ma trận

- Trang 186

p là đầu ra thực sự `vì đầu ra thực sự vì` => p là đầu ra thực sự `vì`

lần lượt `là là` phần tử thứ j => lần lượt `là` phần tử thứ j

- Trang 188

Trong đó, gradient `theo theo` từng cột => Trong đó, gradient `theo` từng cột

- Trang 196

Bước này được `gói` là feedforward => Bước này được `gọi` là feedforward

- Trang 197

`weghts` và biases => `weights` và biases

- Trang 201

ReLU được chứng `mình` => ReLU được chứng `minh`

- Trang 209

Sự phức tạp này ảnh hướng tới hai `khia` cạnh => Sự phức tạp này ảnh hướng tới hai `khía` cạnh 

- Toàn sách:

Chưa sử dụng thống nhất `neural network` với `neural networks`

Font tiếng việt bị lỗi khi thực hiện copy trên PC/MAC, hoặc khi convert file PDF sang PRC để đọc trên máy đọc sách Kindle

**Comment by tiepvupsu** [comment]
Đã sửa hết lỗi trong comment gần nhất của @ZuzooVn. Font tiếng Việt anh sẽ sửa lại sau khi ra sách giấy. 

Cảm ơn em.

**Comment by phanhaiquang** [comment]
Trang 68: "tốc độc của xe" --> "tốc độ của xe"
![image](https://user-images.githubusercontent.com/1145845/37553302-dcfba652-29f7-11e8-93db-735e21874289.png)

**Comment by nhanph** [comment]
Em xin góp ý thêm:
Trang 23: "khả nghich" => khả nghịch
![screen shot 2018-05-17 at 4 03 26 am](https://user-images.githubusercontent.com/15070161/40164412-71ca664c-5987-11e8-8ef4-47d27b3facff.png)
